modules/caddytls/connpolicy.go

```diff
@@ -280,6 +293,30 @@ func (p *ConnectionPolicy) buildStandardTLSConfig(ctx caddy.Context) error {
 		}
 	}
 
+	if p.InsecureSecretsLog != "" {
+		filename, err := caddy.NewReplacer().ReplaceOrErr(p.InsecureSecretsLog, true, true)
+		if err != nil {
+			return err
+		}
+		filename, err = filepath.Abs(filename)
+		if err != nil {
+			return err
+		}
+		logFile, _, err := secretsLogPool.LoadOrNew(filename, func() (caddy.Destructor, error) {
+			w, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0600)
+			return destructableWriter{w}, err
+		})
+		if err != nil {
+			return err
+		}
+		ctx.OnCancel(func() { _, _ = secretsLogPool.Delete(filename) })
```

Comment:
All this just makes sure that through config reloads, only one instance of each log file is opened at a time. This isn't safe for multiple TLS configs that write to the same file, but you know what, this feature better not be used by busy servers in production -- only for testing and debugging! -- so I'm not too worried about it.

It also makes sure the log file is closed whenever the next config (if any) doesn't use it anymore.